File: Solutions/Easy/3379. Transformed Array/Solution.py
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        result: List[int] = []
        n: int = len(nums)
        for i in range(0, n):
            move: int = i + nums[i]
            result.append(nums[move % n])
        return result

File: Solutions/Easy/3379. Transformed Array/Solution.cs
public class Solution {
    public int[] ConstructTransformedArray(int[] nums) {
        int n = nums.Length;
        int[] result = new int[n];
        for(int i = 0; i < n; i++){
            int move = i + nums[i];
            if(move < 0) move = move % n + n;
            result[i] = nums[move % n];
        }
        return result;
    }
}